pytradfri/aiocoap_api.py

```diff
@@ -0,0 +1,88 @@
+"""Coap implementation using aiocoap."""
+import json
+import logging
+
+import asyncio
+from aiocoap import Message, Context, GET, PUT
+from aiocoap.transports.tinydtls import PSK_STORE
+from pytradfri.command import Command
+from .parser import process_output
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def api_factory(host, security_code):
+    """Generate a request method."""
+
+    PSK_STORE[b'Client_identity'] = security_code.encode('utf-8')
```

Comment:
I've rebased my branch against the upstream tinydtls branch, but I'm confused about a few things;

https://github.com/chrysn/aiocoap/blob/tinydtls/aiocoap/transports/tinydtls.py#L56 - is this a temporary hack?

https://github.com/chrysn/aiocoap/blob/tinydtls/aiocoap/protocol.py#L525 - this and DTLSSecurityStore are created inline, and the only way to access the transport_protocols is from the context?

I can't see a clean way of providing auth details to the context so they're passed down to the DTLSSecurityStore?